index.bs

```diff
@@ -10056,35 +10174,60 @@ interface AudioWorkletProcessor {
+		named <code>processorerror</code> at the {{AudioWorkletNode}}
+		on the <a>control thread</a> with the relevant information about
+		the error.
+
 		<div algorithm="AudioWorkletProcessor()">
-			1. Compare the value of <a href="https://www.ecma-international.org/ecma-262/6.0/#sec-built-in-function-objects">NewTarget</a>
-				with the <a href="https://www.ecma-international.org/ecma-262/6.0/#sec-execution-contexts">active function object</a>;
-				if the two are equal, throw a {{TypeError}}.
-
-				Note: This check prevents the invocation of the constructor
-				directly from JavaScript. The interface object may only
-				be called internally by the UA.
-
-			1. Initialize the {{AudioWorkletProcessor}} using the
-				contents of the {{AudioWorkletNode/AudioWorkletNode()/options!!argument}} dictionary that was
-				passed to the constructor for {{AudioWorkletNode}}. The
-				contents of this dictionary will have been serialized and
-				deserialized according to the algorithm for <a href=
-				"#instantiation-of-AudioWorkletNode-and-AudioWorkletProcessor"> instantiating an AudioWorkletProcessor</a>.
-
-				Note: The base implementation of {{AudioWorkletProcessor}}
-				does not examine the values in the {{AudioWorkletProcessor/AudioWorkletProcessor()/options!!argument}}
-				dictionary. However, processor classes that extend
-				{{AudioWorkletProcessor}} will use this dictionary to
-				initialize themselves based on the options relevant to
-				their specific node type.
-
-			1. Set {{[[node reference]]}} to `null` and {{[[callable process]]}} to
-				`false`.
+			This constructor is invoked by processing a control
+			message queued by the {{AudioWorkletNode()|constructor}}
+			of the associated {{AudioWorkletNode}}. This control
+			message carries the [=processor construction data=] that
+			consists of:
+			<var>nodeName</var>,
+			<var>serializedProcessorPort</var>,
+			<var>serializedOptions</var>,
+			and <var>node</var>.
+
+			1. If there is no [=processor construction data=]
+				passed from the associated {{AudioWorkletNode}},
+				throw a {{TypeError}}.
+
+			1. Let <var>processorPort</var> be
+				[$StructuredDeserializeWithTransfer$](<var>serializedProcessorPort</var>,
+				the current Realm).
+
+			1. Let <var>options</var> be
+				[$StructuredDeserialize$](<var>serializedOptions</var>,
+				the current Realm).
+
+			1. Let <var>processorCtor</var> be the result of looking
+				up <var>nodeName</var> on the
+				{{AudioWorkletGlobalScope}}'s
+				<a>node name to processor constructor map</a>.
+
+			1. Let <var>processor</var> be the result of
+				Construct(<var>processorCtor</var>, « <var>options</var> »).
+
+			1. Set <var>processor</var>’s port to <var>processorPort</var>.
+
+			1. Set <var>processor</var>'s {{[[node reference]]}} to
+				<var>node</var>.
+
+			1. Set {{[[callable process]]}} to `true`.
+
+			1. Set <var>node</var>'s <a>processor reference</a> to
```

Comment:
Not relevant for this CL, but maybe we should make "processor reference" an internal slot like callable process, node reference, and so on.

Reply:
ACK